### PR TITLE
 ENH: Synchronize changes between module widget and home widget

### PR DESCRIPTION
**Pull request by mohammadrashid0917**
These changes make it so that changes in the module widget (viewable in Slicer) are reflected in the home widget (viewable in VR) and vice versa

### CODE REVIEW

**Comment by mohammadrashid0917** [comment]
I'm not sure why the Add VR optimized Qt Style Sheet commit is showing up here. It only shows one commit with that name in my git log, not two. That commit should already be merged into this branch

If it's a problem could you try squashing as you merge? I'll keep trying to figure it out from my end until then

**Comment by cpinter** [comment]
Please change the commit's name. It should contain what the change is, and not the activity you did to reach it. "ENH: Make VR home widget controls functional"

I think it shows up because you sent the PR to my fork and not the original repo.

File: VirtualReality/Widgets/qMRMLVirtualRealityHomeWidget.cxx [hunk 1]
```diff
@@ -130,19 +135,157 @@ void qMRMLVirtualRealityHomeWidget::setVirtualRealityViewNode(vtkMRMLVirtualReal
...
+  d->FlySpeedSliderWidget->setEnabled(vrViewNode != nullptr);
+  d->FlySpeedSliderWidget->blockSignals(wasBlocked);
+
+  /*
+  bool wasBlocked = d->LockMagnificationCheckBox->blockSignals(true);
+  d->LockMagnificationCheckBox->setChecked(vrViewNode != nullptr && vrViewNode->);
+  d->LockMagnificationCheckBox->setEnabled(vrViewNode != nullptr);
+  d->LockMagnificationCheckBox->blockSignals(wasBlocked);
+  */
+  //Magnification lock of view node not implemented yet
```
Comment (cpinter) [inline VirtualReality/Widgets/qMRMLVirtualRealityHomeWidget.cxx]: This is a todo, please make it so: //TODO: Magnification...

File: VirtualReality/Widgets/qMRMLVirtualRealityHomeWidget.cxx [hunk 2]
```diff
@@ -130,19 +135,157 @@ void qMRMLVirtualRealityHomeWidget::setVirtualRealityViewNode(vtkMRMLVirtualReal
...
+
+//-----------------------------------------------------------------------------
+void qMRMLVirtualRealityHomeWidget::onMotionSensitivityChanged(double percent)
+{
+  Q_D(qMRMLVirtualRealityHomeWidget);
+  vtkMRMLVirtualRealityViewNode* vrViewNode = d->VirtualRealityViewNode;
+
+  if (!vrViewNode)
+  {
+    qCritical() << Q_FUNC_INFO << " Failed: view node is null";
```
Comment (cpinter) [inline VirtualReality/Widgets/qMRMLVirtualRealityHomeWidget.cxx:173]: You also need to return here obviously, otherwise the next line results in an immediate crash.

File: VirtualReality/Widgets/qMRMLVirtualRealityHomeWidget.cxx [hunk 3]
```diff
@@ -130,19 +135,157 @@ void qMRMLVirtualRealityHomeWidget::setVirtualRealityViewNode(vtkMRMLVirtualReal
...
+
+//-----------------------------------------------------------------------------
+void qMRMLVirtualRealityHomeWidget::onFlySpeedChanged(double speedMps)
+{
+  Q_D(qMRMLVirtualRealityHomeWidget);
+  vtkMRMLVirtualRealityViewNode* vrViewNode = d->VirtualRealityViewNode;
+
+  if (!vrViewNode)
+  {
+    qCritical() << Q_FUNC_INFO << " Failed: view node is null";
```
Comment (cpinter) [inline VirtualReality/Widgets/qMRMLVirtualRealityHomeWidget.cxx:188]: Same here, please fix everywhere.

File: VirtualReality/Widgets/qMRMLVirtualRealityHomeWidget.cxx [hunk 4]
```diff
@@ -130,19 +135,157 @@ void qMRMLVirtualRealityHomeWidget::setVirtualRealityViewNode(vtkMRMLVirtualReal
...
+  }
+  qMRMLVirtualRealityView* vrView = vrModule->viewWidget();
+  if (!vrView)
+  {
+    return;
+  }
+  vrView->updateViewFromReferenceViewCamera();
+}
+*/
+//This member function won't work unless qSlicerVirtualRealityModule and qMRMLVirtualRealityView are included
```
Comment (cpinter) [inline VirtualReality/Widgets/qMRMLVirtualRealityHomeWidget.cxx]: Make it TODO

**Comment by cpinter** [comment]
@mohammadrashid0917 I added some comments to the commit. Also please close this PR and create another one for KitwareMedical/SlicerVirtualReality instead of cpinter/SlicerVirtualReality

File: VirtualReality/Widgets/qMRMLVirtualRealityHomeWidget.h [hunk 5]
```diff
@@ -39,7 +39,7 @@ class qMRMLVirtualRealityHomeWidgetPrivate;
 class Q_SLICER_QTMODULES_VIRTUALREALITY_WIDGETS_EXPORT qMRMLVirtualRealityHomeWidget : public qMRMLWidget
 {
   Q_OBJECT
-  QVTK_OBJECT
+    QVTK_OBJECT
```
Comment (cpinter) [inline VirtualReality/Widgets/qMRMLVirtualRealityHomeWidget.h]: Please undo this change and make QVTK_OBJECT the same indentation as Q_OBJECT
